### PR TITLE
Set different default display style for different commands

### PR DESCRIPTION
**Pull request by Tom-DaLin**
There are two improvements in this submission. The first improvement is to set the default display output format for some commands.
For example, the `dashboard` command outputs graph format, and the `metrics list` command outputs table format.
The second improvement is the addition of command association. When the user enters a wrong command, the command that the user may need will be displayed.
Snipaste_2021-07-21_12-54-43
Snipaste_2021-07-21_12-55-12

### CODE REVIEW

**Comment by kezhenxu94** [comment]
Fix the CI please

**Review by kezhenxu94 — DISMISSED** [review]
LGTM basically. Wait for tests results and @fgksgf ’s confirmation. 

> The second improvement is the addition of command association. When the user enters a wrong command, the command that the user may need will be displayed.

This is extremely useful! Did you take reference of other popular tools like `kubectl` how they do this?

**Comment by Tom-DaLin** [comment]
> LGTM basically. Wait for tests results and @fgksgf ’s confirmation.
> 
> > The second improvement is the addition of command association. When the user enters a wrong command, the command that the user may need will be displayed.
> 
> This is extremely useful! Did you take reference of other popular tools like `kubectl` how they do this?

Through some information on the Internet, I learned how to achieve this function, but I am not sure how other tools do it.

**Comment by wu-sheng** [comment]
CI is still failing, please recheck.

**Comment by Tom-DaLin** [comment]
> CI is still failing, please recheck.

Excuse me, what should I do?

**Comment by wu-sheng** [comment]
image

Click details, and check the logs. CI failure means your latest codes don't pass the checks, maybe code style, license headers requirement, unit tests, etc.

**Comment by fgksgf** [comment]
> > LGTM basically. Wait for tests results and @fgksgf ’s confirmation.
> > > The second improvement is the addition of command association. When the user enters a wrong command, the command that the user may need will be displayed.
> > 
> > 
> > This is extremely useful! Did you take reference of other popular tools like `kubectl` how they do this?
> 
> Through some information on the Internet, I learned how to achieve this function, but I am not sure how other tools do it.

@Tom-DaLin  The screenshot I showed to you is `kubectl`, so the second improvement imitates how `kubectl` does when enter the wrong command.

image

File: pkg/display/display.go [hunk 1]
```diff
@@ -56,3 +67,22 @@ func Display(ctx *cli.Context, displayable *d.Displayable) error {
...
+func getDisplay(fullName string) string {
+	// The variable displaystyle sets the input command and the corresponding output format.
+	var style = [][]string{{"dashboard global", "graph"},
+		{"dashboard global-metrics", "graph"},
+		{"metrics top", "table"},
+		{"metrics linear", "graph"},
+		{"metrics list", "table"},
+		{"service list", "table"},
+		{"t", "graph"},
+		{"trace", "graph"}}
```
Comment (fgksgf) [inline pkg/display/display.go:81]: I suggest moving it out of the function and use `map[string]string` to maintain the display style. The above comment seems to be inconsistent.

File: pkg/display/display.go [hunk 2]
```diff
@@ -56,3 +67,22 @@ func Display(ctx *cli.Context, displayable *d.Displayable) error {
 		return fmt.Errorf("unsupported display style: %s", displayStyle)
 	}
 }
+
+// getDisplay gets the default display settings.
+func getDisplay(fullName string) string {
```
Comment (fgksgf) [inline pkg/display/display.go:72]: `getDisplayStyle` may be more explicit

**Review by fgksgf — CHANGES_REQUESTED** [review]
LGTM basically, just some nits.
Good job!

**Comment by fgksgf** [comment]
@Tom-DaLin  CI failed probably because the change of default display style, explicitly specify the display format for these commands with `--display=json`, please.
https://github.com/apache/skywalking-cli/blob/a9d05e8684098884badcd08ae4ef2cc4516c3c26/scripts/test_commands.sh#L39-L68

**Comment by kezhenxu94** [comment]
> @Tom-DaLin CI failed probably because the change of default display style, you need explicitly specify the display format for these commands with `--display=json`.
> 
> Does this need to file a new PR? @kezhenxu94

Both are OK to me, if you or @Tom-DaLin wants to file a new PR to set the display styles for the tests first.

**Comment by Tom-DaLin** [comment]
> @Tom-DaLin CI failed probably because the change of default display style, you need explicitly specify the display format for these commands with `--display=json`.
> 
> Does this need to file a new PR? @kezhenxu94
> 
> https://github.com/apache/skywalking-cli/blob/a9d05e8684098884badcd08ae4ef2cc4516c3c26/scripts/test_commands.sh#L39-L68


But if I set `--display` to json by default, then I will not be able to tell whether the user has set `--display` flag.

**Comment by kezhenxu94** [comment]
> > @Tom-DaLin CI failed probably because the change of default display style, you need explicitly specify the display format for these commands with `--display=json`.
> > Does this need to file a new PR? @kezhenxu94
> > https://github.com/apache/skywalking-cli/blob/a9d05e8684098884badcd08ae4ef2cc4516c3c26/scripts/test_commands.sh#L39-L68
> 
> But if I set `--display` to json by default, then I will not be able to tell whether the user has set `--display` flag.

@Tom-DaLin what we mean is to set the display style explicitly in the test script, there is nothing to do with how the users will be using, 

The scripts 👇 is only for CI testing

https://github.com/apache/skywalking-cli/blob/a9d05e8684098884badcd08ae4ef2cc4516c3c26/scripts/test_commands.sh#L39-L68

**Review by fgksgf — APPROVED** [review]
LGTM

**Pull request merged**